### PR TITLE
Use local os storage to store master key and encrypt saved password with it before saving to pgadmin db instead of one entry for each saved password.#7076

### PR DESCRIPTION
**Pull request by yogeshmahajan-1903**

### CODE REVIEW

**Review by adityatoshniwal — CHANGES_REQUESTED** [review]
We should also cleanup a key of format `pgAdmin4-desktop-user-aditya.toshniwal14@gmail.com`. I don't think its required anymore.

Apart from that, code looks good to me.

File: web/pgadmin/utils/master_password.py [hunk 1]
```diff
@@ -23,29 +28,44 @@ def get_crypt_key():
...
 
+def get_master_password_key_from_os_secret():
+    master_key = None
+    try:
+        # Try to get master key is from local os storage
+        master_key = keyring.get_password(
+            KEY_RING_SERVICE_NAME, KEY_RING_USER_NAME)
+        # if not master_key:
+        #     # if master_key is none then auto generate master_key
+        #     master_key = secrets.token_urlsafe(12)
```
[the remark this answers is not in the document]
Comment (yogeshmahajan-1903) [reply web/pgadmin/utils/master_password.py:51]: Done.

File: web/pgadmin/authenticate/webserver.py [hunk 2]
```diff
@@ -24,6 +24,8 @@
 from pgadmin.utils.csrf import pgCSRFProtect
 from flask_security.utils import logout_user
 
+from ..utils.master_password import set_crypt_key
```
[the remark this answers is not in the document]
Comment (yogeshmahajan-1903) [reply web/pgadmin/authenticate/webserver.py:27]: Done.

**Comment by yogeshmahajan-1903** [comment]
> We should also cleanup a key of format `pgAdmin4-desktop-user-aditya.toshniwal14@gmail.com`. I don't think its required anymore.
> 
> Apart from that, code looks good to me.

Done

**Pull request merged**